testing/tests/DevExpress.ui.widgets/treeViewParts/searching.js

```diff
@@ -0,0 +1,425 @@
+/* global DATA, initTree */
+
+import TreeViewTestWrapper from "../../../helpers/TreeViewTestHelper.js";
+import $ from "jquery";
+const createInstance = (options) => new TreeViewTestWrapper(options);
+
+QUnit.module("searching");
+
+['dataSource', 'items'].forEach((optionName) => {
+    QUnit.test(`Search works even with loop/cycle in ${optionName} option`, function(assert) {
+        let options = { dataStructure: "plain", rootValue: 1, searchEnabled: true };
+        options[optionName] = [
+            { id: 1, text: "item1", parentId: 2, selected: false, expanded: false },
+            { id: 2, text: "item1_1", parentId: 1, selected: false, expanded: false },
+            { id: 3, text: "item2", parentId: 1, selected: false, expanded: false }];
```

Comment:
It's easier to understand synchronized names:
- "item1" is the first item of the first level
- "item1_1" is a child of "item1" and it is first (because "_1")
- "item2" is the second items of the first level (similar to "item1"), but it is a second child of the "item1" node. I expected "item1_2" as the second child of the "item1" node

Reply:
Fixed